homeassistant/components/wled/select.py

```diff
@@ -99,6 +99,34 @@ async def async_select_option(self, option: str) -> None:
         """Set WLED segment to the selected preset."""
         await self.coordinator.wled.preset(preset=option)
 
+class WLEDEffectSelect(WLEDEntity, SelectEntity):
+    """Defined a WLED Effect select."""
+
+    _attr_translation_key = "effect"
+
+    def __init__(self, coordinator: WLEDDataUpdateCoordinator) -> None:
+        """Initialize WLED ."""
+        super().__init__(coordinator=coordinator)
+
+        self._attr_unique_id = f"{coordinator.data.info.mac_address}_effect"
+        self._attr_options = [effect.name for effect in coordinator.data.effects]
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return len(self.coordinator.data.effects) > 0 and super().available
+
+    @property
+    def current_option(self) -> str | None:
+        """Return the current selected effect."""
+        if not isinstance(self.coordinator.data.state.effect, Effect):
```

Comment:
What could effect ever be?

Reply:
In WLED effect is the current light routine.  It could be solid or one of any choice the controller supports.  These are already exposed in the light via the state_attr(entity_id.effect_list).